civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -854,12 +854,100 @@ function civihr_employee_portal_views_default_views() {
 
     if (isset($view)) {
       $views[$view->name] = $view;
+      civihr_employee_portal_fix_custom_fields_mapping($view);
     }
   }
 
   return $views;
 }
 
+/**
+ * To fix mapping on Views import of CiviCRM custom fields
+ *
+ * @param view $view
+ *     the view that is being imported / recreated
+ */
+function civihr_employee_portal_fix_custom_fields_mapping($view) {
+  global $databases;
+  // iterating displays in view
+  foreach ($view->display as $display) {
+    // iterating options on display: fields, filters, arguments
+    foreach ($display->display_options as &$display_option) {
+      if (!is_array($display_option)) {
+        continue;
+      }
+      foreach ($display_option as &$option) {
+        // only continue if this option is about a field
+        if (!isset($option['field'])) {
+          continue;
+        }
+        $option_id_parts = explode('_', $option['field']);
+        $final_id_part = array_pop($option_id_parts);
+        // it must contain underscores and a number at the end to be checked
+        if (!$option_id_parts > 0 || !intval($final_id_part)) {
+          continue;
+        }
+        // checking if this table is part of the civicrm views integration
+        if ( isset($databases['default']['default']['prefix'][$option['table']]) ) {
+          $prefix_table = $databases['default']['default']['prefix'][$option['table']];
+          $table = $prefix_table.$option['table'];
+          $columns = civihr_employee_portal_get_columns_from_table($table);
+          // if the table where the current field must exists does not have the field
+          // it could be fixed...
+          if (!isset($columns[$option['field']])) {
+            $field_name_base = civihr_employee_portal_field_name_without_sequential($option['field']);
+            // searching field by base name in columns
+            $pos = array_search($field_name_base, $columns);
+            // if found then replace
+            if ($pos) {
+              $option['field'] = array_search($columns[$pos], $columns);
+            }
+          }
+        }
+      }
+    }
+  }
+}
```

Comment:
This entire function is really difficult to read given it has so many levels of nested `foreach` loops and ifs. It can definitely be split into smaller functions. Please try to do that

Reply:
@davialexandre I move everything to a class and separate code with multiple functions.